gems/sorbet-runtime/test/types/edge_cases.rb

```diff
@@ -183,6 +183,52 @@ class << self
         assert_equal(:foo, klass.bar)
         assert_equal(:foo, klass.foo)
       end
+
+      it 'handles method reference without sig' do
+        klass = Class.new do
+          extend T::Sig
+          extend T::Helpers
+          def self.foo
+            [-1, -2, -3].map(&method(:bar))
+          end
+          def self.bar(x)
+            -x
+          end
+        end
+        assert_equal([1, 2, 3], klass.foo)
```

Comment:
Is it worth adding assertions to verify that we do, or don't, end up executing runtime checking here?

I'm not sure if I'd expect runtime checks to be executed in the `with runtime checking` case; they certainly shouldn't be in the other two cases.

Reply:
Good idea. I added an assertion for the two other tests. For this one I'm not sure what I can show more as we don't have signature, nothing is checked anyway.